[FIRRTL][LowerToHW] Rename,reorder memory ports. NFC.

prithayan opened the pull request:
Rename and reorder the memory ports when FIRRTL memories. This PR makes sure 
the ports match with the Scala FIRRTL Compiler, to be compatible with other 
memory generators.

richardxia reviewed (approved):
I compared this with the output of SFC, and looks like it matches SFC's naming convention and ordering of ports.

prithayan commented:
Thanks @richardxia

richardxia commented:
Sorry about that! I probably need to familiarize myself with the code better so that I understand how things are connected. 🤐

prithayan commented:
Thanks @seldridge for catching this and reverting the commit. 
I added a firtool test for memory, and updated `HWMemSimImpl`. 
Also, thanks @richardxia for reviewing and don't worry, we learn something new here everytime  :)
